tuxemon/core/states/world/worldstate.py

```diff
@@ -634,12 +631,17 @@ def get_exits(self, position, collision_map=None, skip_nodes=None):
                 ("up", (position[0], position[1] - 1)),
                 ("left", (position[0] - 1, position[1])),
         ):
+            # if exits are defined make sure the neighbor is present there
+            if exits and not neighbor in exits:
+                continue
+
+            # check if the neighbor region is present in skipped nodes
             if neighbor in skip_nodes:
                 continue
 
             # We only need to check the perimeter,
             # as there is no way to get further out of bounds
-            if position[0] in self.invalid_x or position[1] in self.invalid_y:
+            if neighbor[0] in self.invalid_x or neighbor[1] in self.invalid_y:
```

Comment:
why is this changed?

Reply:
The out of bounds check is an entirely different bug. This PR fixes two separate issues: Collision tiles with `exit_to` letting the player walk into neighboring solids, and map bounds not being checked properly. I was going to make a different PR for each one, but it's the exact same piece of code so I did them both here to avoid Git conflicts.

Regarding the `exit_to` check I explained in the review below how and why I changed that. As far as the map boundary fix goes, it was broken in two places: First it was checking the player's current position, not where they're attempting to walk toward... the modification quoted here fixes that part. Second issue was that the + 1 added to invalid_x and invalid_y further down added an extra tile to the right and bottom, since map tiles are counted from 0 so for instance 36 is actually 35... this allowed the player to walk one extra tile out of bounds in those directions.

I already tested those modifications individually to make sure they each do what I understood they do, by walking into each edge of two differently sized maps. My tests showed that each change was necessary to fix a particular problem.